x-pack/platform/plugins/shared/fleet/server/types/rest_spec/fleet_server_policy_config.ts

```diff
@@ -39,6 +75,34 @@ export const PutFleetServerHostRequestSchema = {
     is_default: schema.maybe(schema.boolean({ defaultValue: false })),
     is_internal: schema.maybe(schema.boolean()),
     proxy_id: schema.nullable(schema.string()),
+    secrets: schema.maybe(
+      schema.object({
+        ssl: schema.maybe(
+          schema.object({
+            key: schema.maybe(secretRefSchema),
+            es_key: schema.maybe(secretRefSchema),
+          })
+        ),
+      })
+    ),
+    ssl: schema.maybe(
```

Comment:
we could extract the common fields here too, and use them like this: https://github.com/elastic/kibana/blob/c44580f38a7143b1fb00f5adf8777e9b2b3ae51e/x-pack/platform/plugins/shared/fleet/server/types/rest_spec/fleet_proxies.ts#L40